solr/core/src/java/org/apache/solr/handler/designer/SchemaDesignerConfigSetHelper.java

```diff
@@ -168,24 +167,12 @@ Map<String, Object> analyzeField(String configSet, String fieldName, String fiel
   }
 
   List<String> listCollectionsForConfig(String configSet) {
-    final List<String> collections = new ArrayList<>();
-    Map<String, ClusterState.CollectionRef> states =
-        zkStateReader().getClusterState().getCollectionStates();
-    for (Map.Entry<String, ClusterState.CollectionRef> e : states.entrySet()) {
-      final String coll = e.getKey();
-      if (coll.startsWith(DESIGNER_PREFIX)) {
-        continue; // ignore temp
-      }
-
-      try {
-        if (configSet.equals(e.getValue().get().getConfigName()) && e.getValue().get() != null) {
-          collections.add(coll);
-        }
-      } catch (Exception exc) {
-        log.warn("Failed to get config name for {}", coll, exc);
-      }
-    }
-    return collections;
+    return zkStateReader()
+        .getClusterState()
+        .collectionStream()
```

Comment:
Performance regression: this forces reading all `state.json` of all collections.
I think `getCollectionNames()` should be part of this PR and used here instead.

Reply:
You are saying this because of the collection name filter that was there.  As I indicated in my self-PR review, I think that was bogus/erroneous.

Reply:
My comment is unrelated to the name filter.
This code collects collection names. The change forces it to read the `state.json` of the collections even though it doesn't need the additional info. The behavior also changes, partially created collections will no longer be considered when previously they were.

I don't know what this specific class does with the collection name, but since this PR deprecates a method without providing an alternate way of achieving the same result (with comparable performance), I stand by my comment to add `getCollectionNames()` in this PR.

Reply:
The DocCollection was being read before as well.  It's where the configSet name is.